Libcloud 652 abq 3 driver

lochiiconnectivity opened the pull request:
[LIBCLOUD-652] Abiquo 2.x is no longer supported by the company, I believe all providers are running 3.x which has major changes to its API. I have updated the Abiquo compute driver.

lochiiconnectivity committed:
move abiquo driver to support v3.x

lochiiconnectivity committed:
and now tests functional

lochiiconnectivity committed:
fix style against guidelines

lochiiconnectivity committed:
[LIBCLOUD-652] Update Abiquo compute driver to support version 3.x

Kami commented:
@lochiiconnectivity Thanks.

If all the providers are indeed running API v3, then removing support for v2 is fine. Otherwise, we should support both versions (we do a similar thing with CloudSigma and some other drivers) - can you please verify that?

Kami commented on libcloud/common/abiquo.py:
```diff
@@ -125,10 +125,16 @@ def parse_error(self):
+                # Most of the exceptions only have one error
+                raise LibcloudError(errors[0].findtext('message'))
+            else:
+                raise LibcloudError('unparseable body')
```
Probably better to at least return raw body in the exception?

Otherwise it will be hard to debug it without using LIBCLOUD_DEBUG.

Kami commented on libcloud/compute/drivers/abiquo.py:
```diff
@@ -236,37 +251,38 @@ def ex_populate_cache(self):
-            # link_datacenter
-            self.connection.cache['locations'][loc] = get_href(e_vdc, 'edit')
+            loc = get_href(e_vdc, 'location')
+            if (loc is not None):
```
Minor style thing - parenthesis are unnecessary.

lochiiconnectivity committed:
Fix two minor points:

1. style issue in (libcloud/compute/drivers/abiquo.py)
2. return whole body when a parse error is encountered to aid debugging (libcloud/common/abiquo.py)

lochiiconnectivity commented:
For the last two points, I've pushed 0298c71 which should address these.
with regards to the support for Abiquo 2.x , I will obtain a covering statement from Abiquo shortly, and I'd like to point out that the driver did not support the last versions of 2.x, 2.3, 2.4 and 2.6, these made significant changes to the API making the trunk version incompatible, (for example making JSON the default media type, see note in http://wiki.abiquo.com/display/ABI26/API+Changelog+v2.6).

Kami commented:
@lochiiconnectivity Yeah, I noticed that in the API v3 change log. Didn't know this back-ward incompatible change was already introduced in v2.6.

Let us know when you hear back from Abiquo.

lochiiconnectivity committed:
Empty commit to trigger travis build, which seems to be stuck

Kami commented:
@lochiiconnectivity Any updates from Abiquo?

lochiiconnectivity commented:
Having spoken to Abiquo, they've confirmed 2.x is not officially supported and they've looked across their customer base and can't see anything that would cause an issue if this support (for 2.x) was removed.

Kami commented:
Thanks. Merged into trunk.
